### DISCUSSION TITLE
Feature requests

### DISCUSSION THREAD

**Discussion by Techineer** [NanoVNA-Saver/nanovna-saver#703] [Ideas]
Two requests for future versions:
1. Can it be made possible to save a path for calibration files?   When I first open the program, I have to hunt through my folder tree to find the folder I saved the calibrations in.  It would be nice to be able to either select from a list of previous locations or automatically go to the last used location.

2. Can the Start, stop frequencies and number of sweeps data be stored and automatically recalled for a saved calibration?   If I calibrate for 3.4 to 4.1 MHz with 10 sweeps and save it, I currently need to save the details in the notes for that calibration and manually set the parameters in the app.  It would certainly be much more convenient to have those details automatically be entered in sweep control when the calibration is loaded.  It would also give visual confirmation that the parameters were loaded, whereas currently you have to take that on faith.

Otherwise, thanks for a very useful app!

**Comment by Peter3579** [comment]
Very good suggestions.

The "logarithmic sweep" setting needs also to be restored. Either additional info must go into the calibration file or we need a bit of heuristics to "guess" the right sweep settings from the frequency data of the calibration file.

Additionally, that 2. works, I'd propose:

3. Do not overwrite the start, stop, and sweep settings when connecting to the NanoVNA. Or add a tickbox at the connection dialog where this could be disabled (and store the settings of that tickbox in the local settings file).

**Comment by BlueBit5** [comment]
Feature Request:

While I can see how to save and restore CAL files, I can't find a way to save and restore general program state.
Each time I exit and restart NanoVNAsaver, I loose my markers, number of points, start and stop frequencies, etc.
When using the same type of DUT for several sessions a day, for many days, it gets very tiring to keep re-entering
7 marker details, display settings, number of points, and sweep frequencies, etc.

Most professional instruments have the ability to save and restore STATE (what we call general program configuration).

Sincere Thanks for considering this hugely useful request.   And many more thanks for this fine software!

73,

 Neal

**Comment by VaR63** [comment]
Many thanke for this awesome app, firstly.

All the above features would be very useful. Also, I would appreciate if the same configuration could be preserved across app restarts, for each detail of the "display settings" too. Currently, only the chart type retains its settings.

Also, to accurately represent the insertion loss of a filter, a new chart type, "S21 Insertion loss", should be implemented. This would be identical to "S21 Gain", except the Y-axis values that should be negated (sign cange).
